### ISSUE TITLE
source code manger scan very slow

### ISSUE THREAD

**Issue by JeenHwahn** [microsoft/vscode#227359]
After a recent update, I found that it was slow to open a new folder or rescan git changes after changes were made. Although my folder may be a little big, just .git will be nearly 10G, but before scanning is very fast, soon can list the file version differences.
    Not only git, cpp intellisense is also slow, both it and git scan consume nearly 3 minutes of my memory of 6 to 8 gigabytes, and memory resources are quickly released after scanning.
    This seems to have happened after updating the Git Graph of your repository.

**Comment by JeenHwahn** [comment]
I think it may be because my git directory is too large, vs comes with the source code manager each time to draw the git commit graph takes a long time to traverse. Can you change the format to show only the last 10 commits and click the "show more" button to see more charts

**Comment by lszomoru** [comment]
@JeenHwahn, can you please try setting `scm.showHistoryGraph` to `false` and see if that helps. Thanks!

**Comment by JeenHwahn** [comment]
Oh, thanks. It got better after disabling scm.showHistoryGraph. But I think showHistoryGraph is a very good function, and I still hope to leave it on in the future. I think if you want to see recent records, you can only show the last few changes, otherwise it will consume too many resources

**Comment by lszomoru** [comment]
Yes, the feature has moved to its own dedicated view - "Source Control Graph" view.
